ngraph/python/setup.py

```diff
@@ -237,144 +49,120 @@ def cpp_flag(compiler):
     "ngraph.impl.passes",
 ]
 
-sources = [PYNGRAPH_SRC_DIR + "/" + source for source in sources]
+data_files = []
 
-include_dirs = [PYNGRAPH_SRC_DIR, NGRAPH_CPP_INCLUDE_DIR, PYBIND11_INCLUDE_DIR]
-
-library_dirs = [NGRAPH_CPP_LIBRARY_DIR]
-
-libraries = [NGRAPH_CPP_LIBRARY_NAME, ONNX_IMPORTER_CPP_LIBRARY_NAME]
-
-extra_compile_args = []
-extra_link_args = []
-
-data_files = [
-    (
-        "lib",
-        [
-            os.path.join(NGRAPH_CPP_LIBRARY_DIR, library)
-            for library in os.listdir(NGRAPH_CPP_LIBRARY_DIR)
-            if os.path.isfile(os.path.join(NGRAPH_CPP_LIBRARY_DIR, library))
-        ],
-    ),
-]
+with open(os.path.join(PYNGRAPH_ROOT_DIR, "requirements.txt")) as req:
+    requirements = req.read().splitlines()
 
-ext_modules = [
-    Extension(
-        "_pyngraph",
-        sources=sources,
-        include_dirs=include_dirs,
-        define_macros=[("VERSION_INFO", __version__)],
-        library_dirs=library_dirs,
-        libraries=libraries,
-        extra_compile_args=extra_compile_args,
-        extra_link_args=extra_link_args,
-        language="c++",
-    ),
-]
 
+class CMakeExtension(Extension):
 
-def add_platform_specific_link_args(link_args):
-    """Add linker flags specific for the OS detected during the build."""
-    if sys.platform.startswith("linux"):
-        link_args += ["-Wl,-rpath,$ORIGIN/../.."]
-        link_args += ["-z", "noexecstack"]
-        link_args += ["-z", "relro"]
-        link_args += ["-z", "now"]
-    elif sys.platform == "darwin":
-        link_args += ["-Wl,-rpath,@loader_path/../.."]
-        link_args += ["-stdlib=libc++"]
-    elif sys.platform == "win32":
-        link_args += ["/LTCG"]
+    def __init__(self, name, sources=[]):
+        super().__init__(name=name, sources=sources)
 
 
-class BuildExt(build_ext):
-    """A custom build extension for adding compiler-specific options."""
+class BuildCMakeExt(build_ext):
 
-    def _add_extra_compile_arg(self, flag, compile_args):
-        """Return True if successfully added given flag to compiler args."""
-        if has_flag(self.compiler, flag):
-            compile_args += [flag]
-            return True
-        return False
+    user_options = build_ext.user_options + [
+        ("config=", "c", "Build configuration [Release| Debug]")
+    ]
+    
+    def initialize_options(self):
+        self.config = None
+        super().initialize_options()
+    
+    def finalize_options(self):
+        if self.config not in ["Release", "Debug"]:
+            self.config = "Release"
```

Comment:
Some type of warning would be nice here.

Reply:
- setup.py build only required components: pyngraph and its dependencies (ngraph, onnx_importer).
- if --config option is not specified for  build_ext, Release build configuration will be used by default.

Reply:
So only available options are Release and Debug case sensitive?
What about RelWithDebInfo?
Could you modify setup.py to build only python API?
It should be working like c++ samples(can be build stand-alone or with the rest of the build) IMO.

Reply:
- made option check case insensitive
- added support options  (Release", "Debug", "RelWithDebInfo", "MinSizeRel)
- allowed to build only python API, if ngraph_DIR env. var was provided